utils/measure/measure/util/measure_util.py

```diff
@@ -79,44 +197,129 @@ def take_measurement(
+            exit(1)
+
+        average = mean(measurements)
+        _LOGGER.info("Average measurement: %.3f W", average)
         return average
 
     def initialize_dummy_load(self) -> float:
-        """Get the previously measured dummy load value, or take a new measurement if it doesn't exist"""
+        """Get the previously measured dummy load resistance, or take a new measurement if it doesn't exist"""
 
         dummy_load_file = os.path.join(
             PROJECT_DIR,
-            ".persistent/dummy_load",
+            ".persistent/dummy_load_resistance",
         )
         if os.path.exists(dummy_load_file):
             with open(dummy_load_file) as f:
                 value = float(f.read())
-            _LOGGER.info("Dummy load was already measured before, value: %sW", value)
+            _LOGGER.info("Dummy load was already measured before, value: %s Ω", value)
 
+            print("You need to preheat the dummy load, so it's consumption can stablize.")
+            print("If you're unsure the dummy load is sufficently preheated or you're using a different one, remeasure.")
+            print()
             inquirer = input("Do you want to measure the dummy load again? (y/n): ")
             if inquirer.lower() == "n":
                 self.dummy_load_value = value
                 return self.dummy_load_value
 
+        print()
+        print("Tip: Use a dummy load with constant power consumption. Stick to resistive loads for the best results!")
+        print("Important: Connect only the dummy load to your smart plug—not the device you're measuring.")
+        print()
+        print("The script will now measure the dummy load and continue once it's consumption has stablized.")
+        print("Depending on the dummy load this may take 2 hours.")
+        print()
+        input("Ready to begin measuring the dummy load? Hit Enter ")
+
         self.dummy_load_value = self._measure_dummy_load(dummy_load_file)
         return self.dummy_load_value
 
     def _measure_dummy_load(self, file_path: str) -> float:
         """Measure the dummy load and persist the value for future measurement session"""
-        print()
-        print("Tip: Use a dummy load with constant power consumption. Stick to resistive loads for the best results!")
-        print("Important: Connect only the dummy load to your smart plug—not the device you're measuring.")
-        print("Preheat your dummy load until its temperature stabilizes. This usually takes about 2 hours.")
-        input("Ready to start? Press Enter to begin measuring the dummy load!")
-        average = self.take_average_measurement(1)
+
+        measurements = 20
+        duration = 30
+
+        print("Measuring and checking dummy load... this will take at least %.0f minutes." % (measurements / 60 * duration))
+
+        # Validate power meter is capable of measuring voltage
+        self._validate_voltage_support()
+
+        while True:
```

Comment:
Yeah, this needs a time limit. Guess 4 hours would be good? This gives it a fair chance to reach a point of equilibrium, regardless what kind of resistor is used.

Reply:
Yes seems fine by me.

Reply:
On the other hand: We measure Resistance here. The resistance can't chance indefinitely - so either measurement drops to 0 W eventually, because the resistance is infinity or the fuse will blow, because the resistance is near to 0 :D

Reply:
Yes I agree.